Sources/PackageLoading/RegistryReleaseMetadataSerialization.swift

```diff
@@ -0,0 +1,141 @@
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+@_implementationOnly import Foundation
+import PackageModel
+import PackageSigning
+import TSCBasic
+
+public enum RegistryReleaseMetadataStorage {
+    public static let fileName = ".registry-metadata"
+
+    private static let encoder = JSONEncoder.makeWithDefaults()
+    private static let decoder = JSONDecoder.makeWithDefaults()
+
+    public static func save(_ metadata: RegistryReleaseMetadata, to path: AbsolutePath, fileSystem: FileSystem) throws {
+        let codableMetadata = CodableRegistryReleaseMetadata(metadata)
+        let data = try Self.encoder.encode(codableMetadata)
+        try fileSystem.writeFileContents(path, data: data)
+    }
+
+    public static func load(from path: AbsolutePath, fileSystem: FileSystem) throws -> RegistryReleaseMetadata {
+        let codableMetadata = try Self.decoder.decode(
+            path: path,
+            fileSystem: fileSystem,
+            as: CodableRegistryReleaseMetadata.self
+        )
+        return RegistryReleaseMetadata(codableMetadata)
+    }
+}
+
+private struct CodableRegistryReleaseMetadata: Codable {
+    public let registry: URL
+    public let signature: Signature?
+    public let author: Author?
+    public let description: String?
+    public let licenseURL: URL?
+    public let readmeURL: URL?
+    public let scmRepositoryURLs: [URL]?
+
+    init(_ seed: RegistryReleaseMetadata) {
+        switch seed.source {
+        case .registry(let url):
+            self.registry = url
+        }
+        self.signature = seed.signature.map {
+            switch $0 {
+            case .recognized(let commonName, let organization, let identity):
+                return .recognized(commonName: commonName, organization: organization, identity: identity)
+            case .unrecognized(let commonName, let organization):
+                return .unrecognized(commonName: commonName, organization: organization)
+            }
+        }
+        self.author = seed.metadata.author.map {
+            .init(
+                name: $0.name,
+                emailAddress: $0.emailAddress,
+                description: $0.description,
+                url: $0.url,
+                organization: $0.organization.map {
+                    .init(
+                        name: $0.name,
+                        emailAddress: $0.emailAddress,
+                        description: $0.description,
+                        url: $0.url
+                    )
+                }
+            )
+        }
+        self.description = seed.metadata.description
+        self.licenseURL = seed.metadata.licenseURL
+        self.readmeURL = seed.metadata.readmeURL
+        self.scmRepositoryURLs = seed.metadata.scmRepositoryURLs
+    }
+
+    public enum Signature: Codable {
```

Comment:
ditto ^. I would call this e.g., `Signer`, `SigningEntity`.

Reply:
And have it inside a `Signature` container struct if we think there will be more properties in the future.